backends/dpdk/dpdkHelpers.cpp

```diff
@@ -346,6 +437,166 @@ bool ConvertStatementToDpdk::preorder(const IR::AssignmentStatement *a) {
     return false;
 }
 
+/* This function processes the hash parameters and stores it in
+   "components" which is further used for generating hash instruction
+*/
+void ConvertStatementToDpdk::processHashParams(const IR::Argument* field,
+                                               IR::Vector<IR::Expression>& components) {
+    if (auto params = field->expression->to<IR::Member>()) {
+        auto typeInfo = typemap->getType(params);
+        if (auto typeheader = typeInfo->to<IR::Type_Header>()) {
+            for (auto it : typeheader->fields) {
+                IR::ID name(params->member.name + "." + it->name);
+                auto m = new IR::Member(new IR::PathExpression(IR::ID("h")), name);
+                components.push_back(m);
+            }
+        } else {
+            components.push_back(params);
+        }
+    } else if (auto s = field->expression->to<IR::StructExpression>()) {
```

Comment:
could it also be a ListExpression?

Reply:
No. 'field->expression' would be either of type 'member' or 'StructExpression'.